projects/igniteui-angular/src/lib/combo/combo.common.ts

```diff
@@ -1202,8 +1202,14 @@ export abstract class IgxComboBaseDirective extends DisplayDensityBase implement
         if (this.comboAPI.valueKey === null) {
             return keys;
         }
+
+        const isNaNvalue = (key: any): boolean => (isNaN(key) && key !== undefined && typeof key !== 'string') ?? false;
```

Comment:
if `(isNaN(key) && key !== undefined && typeof key !== 'string')` returns `false` we fallback to `false`. No need of this fallback.